### PR TITLE
Cache song_text and song_icon

### PR DESCRIPTION
**Pull request by JasonLG1979**
Basically if the text or icon hasn't changed don't update it.

### CODE REVIEW

**Comment by JasonLG1979** [comment]
@TingPing We briefly talked about this on the IRC. I believe it was your idea.

File: pithos/pithos.py [hunk 1]
```diff
@@ -879,8 +879,10 @@ def update_song_row(self, song = None):
         if song is None:
             song = self.current_song
         if song:
-            self.songs_model[song.index][1] = self.song_text(song)
-            self.songs_model[song.index][2] = self.song_icon(song) or ""
+            if self.songs_model[song.index][1] != self.song_text(song):
+                self.songs_model[song.index][1] = self.song_text(song)
+            if self.songs_model[song.index][2] != self.song_icon(song) or "":
```
Comment (TingPing) [inline pithos/pithos.py]: Remove the extra `or ""`

Comment (JasonLG1979) [inline pithos/pithos.py]: I guess I don't know which one is extra?

Comment (TingPing) [inline pithos/pithos.py]: The one in the if statement.

**Comment by JasonLG1979** [comment]
Done.

**Comment by JasonLG1979** [comment]
Turns out after testing it by putting a simple "print("update text")" and "print("update icon")" after each if statement the extra or "" is in fact not extra. Without the icon part doesn't work. It still updates the icon even when not needed.

**Pull request closed without merging**